### ISSUE TITLE
Specific data object causes Oj to crash with SIGABRT (in Rails setup)

### ISSUE THREAD

**Issue by raimondasv** [ohler55/oj#417]
to_json crashes with C stack overflow -> SIGABRT in Rails when Oj tries to encode object of specific length and text.

Oj version 3.3.4
OSes - OSX and Centos7
ActiveSupport version: 5.1.2
Ruby version: 2.4.1

To reproduce (setup requires Oj and ActiveSupport gems to be installed). In case you are wondering what kind of text is that - it is a simple text with emojis and German accent letters. Letters are replaced with `*` for the privacy reasons. 

```
require 'oj'
require 'active_support/json/encoding'

Oj::Rails.set_encoder
ActiveSupport::JSON::Encoding.escape_html_entities_in_json = false

text = [
  42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 195, 156, 42, 32, 195, 156, 42, 42,
  42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 13,
  10, 240, 159, 147, 183, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 58, 32, 42, 42, 42, 42, 42, 42, 42,
  42, 42, 42, 32, 13, 10, 240, 159, 147, 141, 32, 42, 42, 42, 42, 195, 188, 42, 42, 42, 42, 42, 32,
  13, 10, 226, 160, 128, 32, 226, 160, 128, 32, 226, 160, 128, 32, 226, 160, 128, 32, 226, 160, 128,
  32, 226, 160, 128, 32, 226, 160, 128, 32, 226, 160, 128, 32, 226, 160, 128, 32, 226, 160, 128, 32,
  226, 160, 128, 32, 226, 160, 128, 32, 226, 160, 128, 32, 226, 160, 128, 32, 226, 160, 128, 32, 226,
  160, 128, 32, 13, 10, 38, 42, 42, 42, 42, 59, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42,
  42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 195, 159, 32, 42, 42, 42, 42, 42, 42,
  42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 42, 32, 42, 42, 42, 32,
  42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 38, 42, 42, 42, 42, 59, 32, 45, 32, 42, 42,
  42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 58, 32, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42,
  42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 42, 42, 46,
  32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 32,
  42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 46, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 46, 46, 32,
  226, 152, 186, 32, 13, 10, 226, 149, 148, 226, 149, 144, 198, 184, 204, 181, 204, 161, 211, 156,
  204, 181, 204, 168, 204, 132, 198, 183, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144,
  226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149,
  144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226,
  149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 151,
  32, 13, 10, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 226, 153,
  165, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42,
  42, 33, 32, 13, 10, 226, 149, 154, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226,
  149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144,
  226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149,
  144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 226, 149, 144, 198, 184, 204, 181,
  204, 161, 211, 156, 204, 181, 204, 168, 204, 132, 198, 183, 226, 149, 144, 13, 10, 13, 10, 226, 128,
  162, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 195, 164, 32, 42, 32, 42, 32, 42, 32, 42,
  32, 42, 32, 42, 32, 42, 32, 42, 32, 226, 128, 162, 32, 13, 10, 42, 42, 42, 42, 42, 42, 32, 42, 42,
  42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 44, 32, 42, 42, 42,
  42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42,
  42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 195, 164, 42,
  42, 42, 42, 46, 32, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42,
  42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32,
  42, 42, 42, 195, 182, 42, 42, 42, 42, 33, 32, 13, 10, 42, 42, 42, 42, 42, 32, 42, 195, 188, 42, 42,
  42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42,
  42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 195, 164, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42,
  32, 42, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42,
  42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42,
  42, 42, 42, 42, 42, 42, 32, 38, 42, 42, 42, 42, 59, 32, 42, 42, 42, 42, 32, 42, 42, 32, 38, 42, 42,
  42, 42, 59, 46, 46, 32, 13, 10, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42,
  42, 32, 42, 42, 42, 42, 42, 58, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42,
  42, 42, 195, 164, 42, 42, 42, 42, 42, 32, 42, 42, 42, 195, 188, 42, 44, 32, 42, 42, 42, 42, 32, 42,
  42, 42, 42, 32, 42, 42, 42, 32, 42, 195, 164, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42,
  42, 42, 195, 182, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42,
  32, 42, 42, 42, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42,
  42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 195, 188, 42, 42, 42, 42, 32,
  42, 42, 42, 42, 42, 42, 42, 46, 13, 10, 40, 33, 41, 32, 42, 42, 32, 42, 42, 42, 42, 32, 195, 188,
  42, 42, 42, 42, 42, 42, 42, 32, 58, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42,
  32, 42, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42,
  32, 45, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42,
  42, 42, 32, 40, 33, 41, 13, 10, 13, 10, 226, 128, 162, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32,
  42, 32, 47, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 47, 32, 42, 32, 42, 32, 42,
  32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 226, 128, 162, 32, 13,
  10, 42, 42, 42, 32, 42, 42, 42, 32, 42, 195, 188, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 45,
  42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 33, 32, 42, 42, 42, 32,
  42, 42, 42, 195, 159, 42, 32, 58, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42,
  42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42,
  42, 42, 42, 32, 40, 32, 42, 42, 42, 195, 188, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42,
  42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 47, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42,
  42, 32, 41, 32, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 46, 32,
  13, 10, 13, 10, 226, 128, 162, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 226, 128,
  162, 32, 13, 10, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42,
  42, 42, 42, 42, 42, 42, 46, 46, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 42,
  32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42,
  42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32,
  42, 195, 182, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 46, 32, 42, 42, 32,
  42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42,
  42, 42, 44, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42,
  42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32,
  42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 32, 40, 32, 42, 46, 42, 46, 32, 42, 42, 42, 32,
  42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 46, 42, 42, 46, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42,
  42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 32, 42,
  42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42,
  42, 42, 32, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32,
  41, 32, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32,
  42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42,
  42, 42, 32, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42,
  32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 44, 32, 42, 42, 42, 42, 42,
  42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 46, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42,
  42, 32, 42, 42, 42, 42, 42, 32, 42, 195, 182, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42,
  42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 195, 164, 42, 42, 42, 42, 32, 42, 42, 42, 42,
  42, 42, 46, 46, 32, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32,
  42, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32,
  42, 42, 42, 42, 42, 42, 42, 32, 45, 32, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42,
  42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42,
  32, 58, 45, 41, 13, 10, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42,
  42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42,
  42, 42, 42, 42, 32, 42, 42, 42, 42, 44, 32, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42,
  42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42,
  42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 32, 42,
  42, 42, 32, 42, 195, 164, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42,
  42, 32, 42, 42, 195, 182, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 195, 188, 42, 42,
  42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42,
  42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42,
  42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42,
  46, 32, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 195, 188,
  42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32,
  42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 46, 32, 13, 10,
  42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42,
  42, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 195, 188, 42, 42, 42,
  42, 42, 32, 42, 42, 42, 42, 33, 32, 13, 10, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42,
  32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42,
  32, 42, 195, 164, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 42,
  42, 42, 42, 32, 42, 42, 42, 42, 46, 32, 40, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 195,
  188, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 42, 32,
  42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 195, 159, 32, 42, 42, 42, 32, 42,
  42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 46, 46, 32, 41, 13, 10, 13,
  10, 226, 128, 162, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 47, 32, 42, 32, 42, 32, 42,
  32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 47, 32, 42, 32, 42, 32, 42, 32, 195, 164, 32, 42, 32,
  42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 42, 32, 226, 128, 162, 32, 13, 10,
  42, 42, 32, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42,
  42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 32, 42, 42, 42,
  32, 42, 195, 164, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42,
  32, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42,
  42, 42, 195, 164, 42, 42, 42, 42, 42, 42, 42, 42, 42, 42, 46, 32, 194, 167, 32, 42, 42, 42, 32, 42,
  42, 42, 46, 42, 32, 42, 42, 42, 32, 13, 10, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42,
  32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 195, 188, 42, 42, 46, 32, 42, 42, 42,
  42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 195, 164, 42,
  42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42, 44, 32, 42, 42, 42, 42, 42, 42, 32,
  42, 42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 32, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42,
  42, 42, 42, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 42, 42,
  46, 32, 13, 10, 42, 42, 42, 32, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42,
  42, 42, 42, 42, 32, 42, 42, 42, 42, 42, 46, 46, 32, 13, 10, 42, 42, 42, 32, 42, 195, 188, 42, 42,
  42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 42, 32, 42, 42, 42, 195, 159, 32, 42, 42, 42, 42,
  32, 42, 42, 42, 42, 42, 42, 42, 42, 33, 32, 58, 45, 41 ]
.pack('c*').force_encoding('UTF-8')

puts text

hash = {
 k1: 'x' * 800,
 k2: text,
}

puts Oj::Rails::Encoder.new.encode(hash)
```
Output:

```
Abort trap: 6
```

System logs:

```
Process:               ruby [63614]
Path:                  /Users/USER/*/ruby
Identifier:            ruby
Version:               0
Code Type:             X86-64 (Native)
Parent Process:        bash [54109]
Responsible:           ruby [63614]
User ID:               501

Date/Time:             2017-08-08 13:53:38.734 +0300
OS Version:            Mac OS X 10.12.6 (16G29)
Report Version:        12
Anonymous UUID:        2ED28AD5-383A-E3B5-8147-3051B0EE2109

Sleep/Wake UUID:       81CC012D-585F-463B-842B-F7C8A7C72D06

Time Awake Since Boot: 53000 seconds
Time Since Wake:       15000 seconds

System Integrity Protection: enabled

Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_CRASH (SIGABRT)
Exception Codes:       0x0000000000000000, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
[63614] stack overflow

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fffeb3d6d42 __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fffeb4c4457 pthread_kill + 90
2   libsystem_c.dylib             	0x00007fffeb33c4bb __abort + 140
3   libsystem_c.dylib             	0x00007fffeb33cd7e __stack_chk_fail + 205
4   oj.bundle                     	0x000000010865dcbc encode + 588
5   ruby                          	0x000000010834a9be vm_call_cfunc + 302
6   ruby                          	0x0000000108334801 vm_exec_core + 11969
7   ruby                          	0x0000000108345514 vm_exec + 116
8   ruby                          	0x00000001081f80f8 ruby_exec_internal + 136
9   ruby                          	0x00000001081f8016 ruby_run_node + 54
10  ruby                          	0x000000010818845f main + 79
11  libdyld.dylib                 	0x00007fffeb2a8235 start + 1

Thread 1:: ruby-timer-thr
0   libsystem_kernel.dylib        	0x00007fffeb3d819e poll + 10
1   ruby                          	0x000000010830dc63 thread_timer + 371
2   libsystem_pthread.dylib       	0x00007fffeb4c193b _pthread_body + 180
3   libsystem_pthread.dylib       	0x00007fffeb4c1887 _pthread_start + 286
4   libsystem_pthread.dylib       	0x00007fffeb4c108d thread_start + 13

Thread 0 crashed with X86 Thread State (64-bit):
  rax: 0x0000000000000000  rbx: 0x0000000000000006  rcx: 0x00007fff57a76c78  rdx: 0x0000000000000000
  rdi: 0x0000000000000307  rsi: 0x0000000000000006  rbp: 0x00007fff57a76ca0  rsp: 0x00007fff57a76c78
   r8: 0x0000000000000000   r9: 0x0000000000000000  r10: 0x0000000008000000  r11: 0x0000000000000206
  r12: 0x00007fff57a76de8  r13: 0x00007fff38615b30  r14: 0x00007ffff41bc3c0  r15: 0x0000000000000000
  rip: 0x00007fffeb3d6d42  rfl: 0x0000000000000206  cr2: 0x00007ffff419e128
  
Logical CPU:     0
Error Code:      0x02000148
Trap Number:     133
```

**Comment by ohler55** [comment]
Looking at it now.

**Comment by ohler55** [comment]
Pushed a fix to master. If tests pass I'll make a release.

**Comment by raimondasv** [comment]
We have compiled this gem from a master and it works ok. No more crashes.

**Comment by ohler55** [comment]
Released 3.3.5.

**Comment by raimondasv** [comment]
@ohler55 thank you!
